agent/lib/kontena/workers/service_pod_worker.rb

```diff
@@ -40,6 +65,7 @@ def apply
           warn "failed to sync #{service_pod.name} at #{service_pod.deploy_rev}: #{error}"
           sync_state_to_master(current_state, error)
         else
+          @container_state_changed = false
```

Comment:
BTW this `ensure_desired_state` -> `@container_state_changed = false` is going to immediately get followed by a `on_container_event` `@container_state_changed = true` if the ensure did something:

```
I, [2017-04-12T07:28:20.429746 #1]  INFO -- Kontena::ServicePods::Creator: service started: redis-1
D, [2017-04-12T07:28:20.501609 #1] DEBUG -- Kontena::Workers::ServicePodWorker: mark as changed on event: create
D, [2017-04-12T07:28:20.503124 #1] DEBUG -- Kontena::Workers::ServicePodWorker: mark as changed on event: start
```

Then the second time around, I'll rerun `ensure_desired_state`, but that shouldn't do anything, so it'll settle down:

```
D, [2017-04-12T07:28:43.772623 #1] DEBUG -- Kontena::Workers::ServicePodWorker: apply with changes
D, [2017-04-12T07:28:43.772901 #1] DEBUG -- Kontena::Workers::ServicePodWorker: state of redis-1: running
D, [2017-04-12T07:28:43.820727 #1] DEBUG -- Kontena::Workers::ServicePodWorker: state is in-sync: running
```

So it's not broken, just... behaves in a slightly surprising way.